interface/new/new_comprehensive_save.php

```diff
@@ -90,6 +91,18 @@
     }
 }
 
+//Save new Weno Pharmacies for Patient
+if($GLOBALS['weno_rx_enable'] && (!empty($_POST['primary_pharmacy']) || 
+    !empty($_POST['alternate_pharmacy']))){
+        
+    $data = array(
+        "primary_pharmacy" => $_POST['primary_pharmacy'],
+        "alternate_pharmacy" => $_POST['alternate_pharmacy']
+    );
+    $pharmacyService = new PharmacyService();
+    $pharmacyService->createWenoPharmaciesForPatient($pid, $data);
+}
```

Comment:
This can be done via a module listener event (there may even already be a event for a patient save)

Reply:
the `$data` array contains properties that are not available in the patient_data table hence it becomes impossible to use the inbuilt patient events.